### DISCUSSION TITLE
Hhy does socket.io automatically disconnect after an hour

### DISCUSSION THREAD

**Discussion by haozero99** [socketio/socket.io#4741] [Q&A]
Hhy does socket.io automatically disconnect after an hour

**Comment by yeuci** [comment]
according to the docs,

1. one HTTP request (either GET or POST) fails (for example, when the server is shutdown)
2. the WebSocket connection is closed (for example, when the user closes the tab in its browser)
3. socket.disconnect() is called on the server-side or on the client-side

they also have a mechanism that checks the connection between the server and client, they use a pingInterval (which you can adjust) to send a packet then the client responds within back with a pong (the pingTimeout is what is used to timeout if over time).

you can use .set to change these, ex:
`js
io.set('connect timeout', 10000);
`

ref: https://socket.io/docs/v4/how-it-works/#:~:text=There%20is%20also%20a%20heartbeat,send%20a%20PONG%20packet%20back.

**Comment by haozero99** [comment]
no effect fixed disconnection every hour

**Comment by darrachequesne** [comment]
Hi! There must be something between your client and your server that closes the connection after a given delay. What is your setup?

That being said, temporary disconnections are expected and must be handled properly by your application.

Reference: https://socket.io/docs/v4/troubleshooting-connection-issues/#problem-the-socket-gets-disconnected

**Comment by haozero99** [comment]
client:"socket.io-client": "^4.6.1" service:  "socket.io": "^4.6.1" transports: ['websocket'], others are default values，It's not like triggering because of delay, because it triggers every hour

**Reply by darrachequesne** [reply]
If it's a fixed delay (exactly 1 hour), then it's something between your client and your server that closes the connection, like a proxy. Is your server hosted behind nginx? Which hosting platform do you use?
